### ISSUE TITLE
add some functionality from pygris

### ISSUE THREAD

**Issue by ctriley** [censusdis/censusdis#305]
I'd like to be able to

1. request block level data, or at the very least, requests the blocks and their geometry
2. I'd like to be able to interact with LODES data (i'm not sure if you can currently do this yet)

https://github.com/walkerke/pygris

**Comment by vengroff** [comment]
Block level data is supported, but there are not many data sets that have block level data. The decennial public law 95.171 is one of the few that do. Here is an example I just tried:

```python
import censusdis.data as ced
import censusdis.maps as cem
from censusdis.states import NJ
from censusdis.counties.new_jersey import HUDSON
from censusdis.datasets import DECENNIAL_PUBLIC_LAW_94_171

gdf_blocks = ced.download(
    DECENNIAL_PUBLIC_LAW_94_171,
    2020,
    ['NAME'],

    state=NJ,
    county=HUDSON,
    block='*',

    with_geometry=True,
    remove_water=True,
)

ax = cem.plot_map(
    gdf_blocks.boundary,
    figsize=(6, 10),
    color="black",
    linewidth=0.25,
)

ax.set_title("Census Blocks in Hudson County, NJ - 2020")
```

It produced this map:

![image](https://github.com/user-attachments/assets/cf87ed0c-ac7f-4c65-a7a0-f113d2e71c56)

I will do a PR with a little notebook with the code in it.

**Comment by vengroff** [comment]
For LODES I will look into it.

**Comment by vengroff** [comment]
Merged in #306

**Comment by vengroff** [comment]
I added some support for LODES behind the familiar `ced.download` interface. It's still alpha and there is more to do, but it works well enough for this [demo notebook](https://github.com/censusdis/censusdis/blob/dev/download-lodes/notebooks/LODES%20Data.ipynb) to work so I created a PR.

Please have a look at the notebook and let me know if you have comments.
